sql/group.go

```diff
@@ -195,6 +214,17 @@ func (n *groupNode) computeAggregates() {
 	for k := range n.buckets {
 		n.currentBucket = k
 
+		if n.having != nil {
+			res, err := n.having.Eval(n.planner.evalCtx)
+			if err != nil {
+				n.err = err
+				return
+			}
+			if res == parser.DNull || !bool(res.(parser.DBool)) {
```

Comment:
Consider exporting parser.getBool and using it here. It should be an error for `res` to not be a boolean, right? Add a test case to that effect as well.